vulnerabilities/importers/nginx.py

```diff
@@ -0,0 +1,190 @@
+        # Chop off the `release-` part here.
+        for index, version in enumerate(self.version_api.cache["nginx/nginx"]):
+            self.version_api.cache["nginx/nginx"][index] = version.replace("release-", "")
+
+    def updated_advisories(self):
+        advisories = []
+        if self.create_etag():
+            self.set_api()
+            data = requests.get(self.url).content
+            advisories.extend(self.to_advisories(data))
+        return self.batch_advisories(advisories)
+
+    def create_etag(self):
+        etag = requests.head(self.url).headers.get("ETag")
+        if not etag:
+            return True
+
+        elif self.url in self.config.etag:
+            if self.config.etag[self.url] == etag:
+                return False
+
+        self.config.etag[self.url] = etag
+        return True
+
+    def to_advisories(self, data):
+        advisories = []
+        soup = BeautifulSoup(data)
+        vuln_list = soup.select("li p")
+
+        # Example value of `vuln_list` :
+        # ['Excessive CPU usage in HTTP/2 with small window updates',
+        #  <br/>,
+        #  'Severity: medium',
+        #  <br/>,
+        #  <a href="http://mailman.nginx.org/pipermail/nginx-announce/2019/000249.html">Advisory</a>,  # nopep8
+        #  <br/>,
+        #  <a href="http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-9511">CVE-2019-9511</a>,
+        #  <br/>,
+        #  'Not vulnerable: 1.17.3+, 1.16.1+',
+        #  <br/>,
+        #  'Vulnerable: 1.9.5-1.17.2']
+
+        for vuln_info in vuln_list:
+            references = []
+            for index, child in enumerate(vuln_info.children):
+                if index == 0:
+                    # type of this child is bs4.element.NavigableString.
+                    # Hence cast it into standard string
+                    summary = str(child)
+                    continue
+
+                #  hasattr(child, "attrs") == False for bs4.element.NavigableString
+                if hasattr(child, "attrs") and child.attrs.get("href"):
+                    link = child.attrs["href"]
+                    references.append(Reference(url=link))
+                    if "cve.mitre.org" in link:
+                        cve_id = child.text
+                    continue
+
+                if "Not vulnerable" in child:
+                    fixed_packages = self.extract_fixed_pkgs(child)
+                    continue
+
+                if "Vulnerable" in child:
+                    vulnerable_packages = self.extract_vuln_pkgs(child)
+                    continue
+
+            advisories.append(
+                Advisory(
+                    cve_id=cve_id,
+                    summary=summary,
+                    impacted_package_urls=vulnerable_packages,
+                    resolved_package_urls=fixed_packages,
+                )
+            )
+
+        return advisories
+
+    def extract_fixed_pkgs(self, vuln_info):
+        vuln_status, version_info = vuln_info.split(": ")
```

Comment:
Using a `.partition()` is more robust as it always does a single split and always return three values making the tuple unpacking safer and more reliable. I tend to prefer it followed by a `.strip()` call.